app/src/main/java/org/fossasia/openevent/app/data/LoginModel.java

```diff
@@ -45,16 +52,29 @@ public boolean isLoggedIn() {
         if(token == null)
             token = utilModel.getToken();
 
-        return token != null && !JWTUtils.isExpired(token);
-    }
+        boolean loggedIn = token != null && !JWTUtils.isExpired(token);
 
-    @Override
-    public void logout() {
-        utilModel.saveToken(null);
+        if(!loggedIn) {
+            //noinspection unchecked
+            databaseRepository.deleteAll(User.class, UserDetail.class)
+                .subscribeOn(Schedulers.io())
+                .subscribe();
+        }
+
+        return loggedIn;
     }
 
-    public void setEventService(EventService eventService) {
-        this.eventService = eventService;
+    @Override
+    public Completable logout() {
+        return Completable.fromAction(() -> {
+            utilModel.saveToken(null);
+            // Bug in foreign key relation, have to manually delete UserDetail
+            //noinspection unchecked
+            databaseRepository.deleteAll(User.class, UserDetail.class)
```

Comment:
Why deleting userdetail explicitely? you have created relationship between user and userdetail right?